### PR TITLE
Use new schema validation library

### PR DESCRIPTION
**Pull request by imbstack**
The major change here is that the new validation library is a bit more strict than the old one. I'll call out any particular changes I noticed in comments on the actual line of the changes, but we should decide whether or not this is an ok change to make before this is shipped.

### CODE REVIEW

File: schemas/authenticate-hawk-request.yml [hunk 1]
```diff
@@ -43,7 +43,9 @@ properties:
       **Note,** order of querystring elements is important.
   host:
     type:                 string
-    format:               hostname
+    oneOf:
```
Comment (imbstack) [inline schemas/authenticate-hawk-request.yml]: ajv is more strict about what it considers a hostname

```js
> var ajv = /^[a-z](?:(?:[-0-9a-z]{0,61})?[0-9a-z])?(\.[a-z](?:(?:[-0-9a-z]{0,61})?[0-9a-z])?)*$/i;
undefined
> var jjv = /^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$/;
undefined
> ajv.test('localhost')
true
> jjv.test('localhost')
true
> ajv.test('127.0.0.1')
false
> jjv.test('127.0.0.1')
true
```

File: test/schemas/create-client-request-no-scopes.json [hunk 2]
```diff
@@ -1,4 +1,4 @@
 {
-  "expires": "2000-02-01 00:00:00",
+  "expires": "2016-03-24T00:48:37.942Z",
```
Comment (imbstack) [inline test/schemas/create-client-request-no-scopes.json]: ajv is also more strict about what it considers to be a valid `date-time`.

File: bin/server.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ var ScopeResolver      = require('../auth/scoperesolver');
 var signaturevalidator = require('../auth/signaturevalidator');
 var taskcluster        = require('taskcluster-client');
 var url                = require('url');
+var validator_         = require('taskcluster-lib-validate');
```
Comment (djmitche) [inline bin/server.js]: could this be `validate` or if that's taken, `lib_validate`, instead?

**Comment by djmitche** [comment]
This looks good, and as we discussed I'm not too worried about the stricter validation in this particular service.  You can change the variable name or not as you see fit :)

File: auth/exchanges.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
-var base      = require('taskcluster-base');
+var Exchanges = require('pulse-publisher');
```
Comment (jonasfj) [inline auth/exchanges.js]: From what I can see we did bump the version...

File: package.json [hunk 5]
```diff
@@ -27,20 +27,24 @@
     "kexec": "^1.2.0",
     "lodash": "^3.9.3",
     "promise": "^7.0.4",
+    "pulse-publisher": "^0.9.1",
```
Comment (jonasfj) [inline package.json]: Shouldn't we bump tc-base instead?

**Pull request merged**